pydantic/schema.py

```diff
@@ -795,7 +795,7 @@ def field_singleton_schema(  # noqa: C901 (ignore complexity)
         f_schema, schema_overrides = get_field_info_schema(field)
         f_schema.update(get_schema_ref(enum_name, ref_prefix, ref_template, schema_overrides))
         definitions[enum_name] = enum_process_schema(field_type)
-    else:
+    elif not hasattr(field_type, '__pydantic_model__'):
```

Comment:
Felt like the easiest way to have a nice schema support was to add `__pydantic_model__` too like we do for `dataclass`